sentences/en/lock_HassGetState.yaml

```
@@ -3,34 +3,42 @@ intents:
   HassGetState:
     data:
       - sentences:
-          - "is <name> {lock_states:state} [in <area>]"
+          - "(is|are) <name> ([<currently>];{lock_states:state} [in <area>])"
```

You can't have a completely optional permutation item, as skipping `<currently>` would match both the optional part at the beginning and at the end, which is inefficient. Also, watch out for spaces before/after optional parts.
```suggestion
          - "(is|are) <name> (<currently>;{lock_states:state}[ in <area>])"
          - "(is|are) <name> {lock_states:state}[ in <area>]"
```

---

I'm not seeing any significant difference in performance when I run benchmarks over 324,000 sentences.

The performance seems to flip flop a bit between the two versions and, even taking worst case difference (0.5s for a benchmark iteration), that comes out to be 0.5, which is one micro second.

With that in consideration, it feels like optimizing for ease of maintenance is more important to giving users a good experience. My worry is not so much making this one change, but doing this for every intent type and then expanding for other common utterances.

Methodology:
```fish
# Number of generated sentences
$ python3 -m hassil.sample ./split_permutations.yaml --areas kitchen --names "front door" | jq -r .text | sort | uniq | wc -l
# Benchmarking with split sentences (your diff)
$ set PERMUTATIONS (python3 -m hassil.sample ./split_permutations.yaml --areas kitchen --names "front door" | jq -r .text)                                                                                                                             [.venv]
  time for i in (seq 1000)
      echo $PERMUTATIONS | hassil ./split_permutations.yaml --areas kitchen --names "front door" 2&> /dev/null
  end

________________________________________________________
Executed in   83.68 secs    fish           external
   usr time   56.92 secs  353.69 millis   56.56 secs
   sys time   14.81 secs  391.86 millis   14.42 secs
# Benchmarking with optional permutations
$ set PERMUTATIONS (python3 -m hassil.sample ./empty_permutations.yaml --areas kitchen --names "front door" | jq -r .text)                                                                                                                             [.venv]
  time for i in (seq 1000)
      echo $PERMUTATIONS | hassil ./empty_permutations.yaml --areas kitchen --names "front door" 2&> /dev/null
  end

________________________________________________________
Executed in   82.60 secs    fish           external
   usr time   56.22 secs  377.65 millis   55.85 secs
   sys time   14.59 secs  398.24 millis   14.20 secs
```

`empty_permutations.yaml`
```yaml
:hassil:Area names: ['kitchen']
language: en
intents:
  HassGetState:
    data:
      - sentences:
          - "(is|are) <name> ([<currently>];{lock_states:state} [in <area>])"
        response: one_yesno
expansion_rules:
  name: "[the] {name}"
  area: "[the] {area}"
  currently: "(currently|presently|right now|at the moment)"
lists:
  lock_states:
    values:
      - in: "[securely ]locked"
        out: "locked"
      - in: "unlocked"
        out: "unlocked"
```

`split_permutations.yaml`
```yaml
language: en
intents:
  HassGetState:
    data:
      - sentences:
          - "(is|are) <name> (<currently>;{lock_states:state} [in <area>])"
          - "(is|are) <name> {lock_states:state} [in <area>]"
        response: one_yesno
expansion_rules:
  name: "[the] {name}"
  area: "[the] {area}"
  currently: "(currently|presently|right now|at the moment)"
lists:
  lock_states:
    values:
      - in: "[securely ]locked"
        out: "locked"
      - in: "unlocked"
        out: "unlocked"
```

---

Well, that's interesting, i must admit. Simply looking at the code structure/complexity says that the split variant should be faster, but i guess that the complexity of other permutation items might have a role here as well.

Either way, I agree that for such minute performance differences, ease of maintenance should be prioritized.

@synesthesiam please take a look at the performance results above and let's discuss if we should change our approach regarding optimization enforcement.

---

Just wanted to see if any more thought has been given to this? I'd like to do another PR generalizing adding some more common values, then start to refactor sentences and generate new structures via an LLM.

It's pretty cool that we can now use an LLM in Assist, but I'm hoping to get things to feel more natural and flexible by letting the LLM generate sentences to match and then performing any actual sentence to intent matching offline.